src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,186 @@
+        return tasks.done(i);
+    }
+
+    private String[] todo(String args) throws IncorrectArgumentException {
+        String[] argv = args.split("/");
+        if (argv[0].equals("")) {
+            throw new IncorrectArgumentException("Oops! Missing required arguments: Task Description");
+        } else if (argv.length > 1) {
+            throw new IncorrectArgumentException("Too many arguments!");
+        }
+        return tasks.addTodo(argv[0]);
+    }
+
+    private String[] deadline(String args) throws IncorrectArgumentException {
+        String[] argv = args.split(" /by ");
+        if (argv[0].equals("")) {
+            throw new IncorrectArgumentException(
+                "Oops! Missing required arguments: "
+                .concat(argv.length == 1 
+                    ? "Task Description, by.." 
+                    : "Task Description"));
+        } else if (argv.length < 2) {
+            throw new IncorrectArgumentException("Oops! Missing required arguments: /by yyyy-MM-dd HHmm");
+        } else if (argv.length > 2) {
+            throw new IncorrectArgumentException("Too many arguments!");
+        }
+
+        String[] resp;
+        try {
+            String[] textDateTime = argv[1].split(" ");
+            LocalDate date = toDate(textDateTime[0]);
+            LocalTime time = toTime(textDateTime[1]);
+            resp = tasks.addDeadline(argv[0], date, time);
+        } catch (DateTimeParseException e) {
+            throw new IncorrectArgumentException("Incorrect Date/Time format!\n     Correct format: yyyy-MM-dd HHmm");
+        }
+
+        return resp;
+    }
+
+    private String[] event(String args) throws IncorrectArgumentException {
+        String[] argv = args.split(" /at ");
+        if (argv[0].equals("")) {
+            throw new IncorrectArgumentException(
+                "Oops! Missing required arguments: "
+                .concat(argv.length == 1 
+                    ? "Task Description, at.."
+                    : "Task Description"));
+        } else if (argv.length < 2) {
+            System.out.println(args);
+            throw new IncorrectArgumentException("Oops! Missing required arguments:\n/at yyyy-MM-dd HHmm");
+        } else if (argv.length > 2) {
+            throw new IncorrectArgumentException("Too many arguments!");
+        }
+
+        String[] resp;
+        try {
+            String[] textDateTime = argv[1].split(" ");
+            LocalDate date = toDate(textDateTime[0]);
+            LocalTime time = toTime(textDateTime[1]);
+            resp = tasks.addEvent(argv[0], date, time);
+        } catch (DateTimeParseException e) {
+            throw new IncorrectArgumentException("Incorrect Date/Time format!\n     Correct format: yyyy-MM-dd HHmm");
+        }
+        return resp;
+    }
+
+    private String[] delete(int i) {
+        return tasks.delete(i);
+    }
+
+    private String[] find(String query) {
+        return tasks.find(query);
+    }
```

Comment:
Even though these functions should not be here, their names can be improved:
1. Function names should be action phrases (X event, bye)
2. Be more descriptive (find/delete what?)